### PR TITLE
Add additional config for enabled JFR build

### PR DESCRIPTION
**Pull request by jhuttana**

### CODE REVIEW

**Comment by karianna** [comment]
@jerboaa Not sure this is the place you want to have this patch?

File: install-rhel6-deps-build-openjdk8.sh [hunk 1]
```diff
@@ -86,7 +86,7 @@ TARBALL_NAME="\${TARBALL_BASE_NAME}-jdk_\${PLATFORM_VERSION}"
 TARBALL_NAME_JRE="\${TARBALL_BASE_NAME}-jre_\${PLATFORM_VERSION}"
 SOURCE_NAME="\${TARBALL_BASE_NAME}-sources_\${TARBALL_VERSION}"
 
-CLONE_URL=https://hg.openjdk.java.net/jdk8u/jdk8u
+CLONE_URL=https://hg.openjdk.java.net/jdk8u/jdk8u-dev
```
Comment (jerboaa) [inline install-rhel6-deps-build-openjdk8.sh]: We don't want to change the clone URL. We'll only merge this PR once the feature has been merged into jdk8u. So please revert this hunk.

**Review by jerboaa — CHANGES_REQUESTED** [review]
@jhuttana Thanks for the PR! Please make similar changes to the build script when BRs are already installed: File `build-openjdk8.sh`.

**Comment by jerboaa** [comment]
> @jerboaa Not sure this is the place you want to have this patch?

@karianna It's the right place. We want to build an additional artifact for 8u262 with JFR enabled while it's still disabled for default config.

File: build-openjdk8.sh [hunk 2]
```diff
@@ -41,10 +41,18 @@ build() {
     MILESTONE="ea"
   fi
 
-  for debug in release slowdebug; do
+  for debug in release release-jfr slowdebug; do
+    if [ "\$debug" == "release-jfr" ]; then
```
Comment (jerboaa) [inline build-openjdk8.sh]: In this script no escaping, please. Should be `$debug` not `\$debug`.

File: build-openjdk8.sh [hunk 3]
```diff
@@ -41,10 +41,18 @@ build() {
...
+       flag="--enable-jfr"
+       dbg_level="release"
+    else
+       flag=""
+       dbg_level="\$debug"
```
Comment (jerboaa) [inline build-openjdk8.sh]: Same.

File: build-openjdk8.sh [hunk 4]
```diff
@@ -41,10 +41,18 @@ build() {
...
+    fi
     bash configure \
        --with-boot-jdk="/usr/lib/jvm/java-1.7.0-openjdk.x86_64" \
-       --with-debug-level="$debug" \
+       "\$flag" \
+       --with-debug-level="\$dbg_level" \
```
Comment (jerboaa) [inline build-openjdk8.sh]: Same.

**Review by jerboaa — CHANGES_REQUESTED** [review]

**Comment by jerboaa** [comment]
@jhuttana So I took this for a spin and there are a couple of thoughts. The build output looks similar to this post patch:

```
$ find build/ -name \*.tar.gz
build/release/images/OpenJDK8U-jdk_x64_linux_8u262b00_ea-debuginfo.tar.gz
build/release/images/OpenJDK8U-jre_x64_linux_8u262b00_ea.tar.gz
build/release/images/OpenJDK8U-jre_x64_linux_8u262b00_ea-debuginfo.tar.gz
build/release/images/OpenJDK8U-jdk_x64_linux_8u262b00_ea.tar.gz
build/OpenJDK8U-sources_8u262b00_ea.tar.gz
build/slowdebug/images/OpenJDK8U-jdk_x64_linux_8u262b00_ea-slowdebug.tar.gz
build/slowdebug/images/OpenJDK8U-jdk_x64_linux_8u262b00_ea-slowdebug-debuginfo.tar.gz
build/release-jfr/images/OpenJDK8U-jdk_x64_linux_8u262b00_ea-debuginfo.tar.gz
build/release-jfr/images/OpenJDK8U-jdk_x64_linux_8u262b00_ea.tar.gz
```

However, we need to change the filename for the JFR-enabled artefacts. They're now named the same as the regular tarballs, namely `OpenJDK8U-jdk_x64_linux_8u262b00_ea.tar.gz`. The JFR artefact should be called `OpenJDK8U-jdk-jfr_x64_linux_8u262b00_ea.tar.gz`. Also, we should try to get JRE's packaged up for the JFR artefacts too: `OpenJDK8U-jre-jfr_x64_linux_8u262b00_ea.tar.gz`

File: build-openjdk8.sh [hunk 5]
```diff
@@ -56,6 +62,13 @@ build() {
       target="images"
     fi
     make LOG_LEVEL=debug CONF=$debug $target
+    if [ "${debug}_" == "release-jfr_" ]; then
+      TARBALL_NAME="${TARBALL_BASE_NAME}-jdk-jfr_${PLATFORM_VERSION}"
```
Comment (jerboaa) [inline build-openjdk8.sh]: I'd prefer if we kept global config variables at the top of the script. So the suggestion would be to define `TARBALL_NAME_JFR` and `TARBALL_NAME_JFR_JRE` and use them in a local variable, say `archive_name` in the script like:

```
archive_name="$TARBALL_NAME"
jre_archive_name="$TARBALL_NAME_JRE"
if [ "${debug}_" == "release-jfr_" ]; then
  archive_name="$TARBALL_NAME_JFR"
  jre_archive_name="$TARBALL_NAME_JFR_JRE"
fi
```

**Review by jerboaa — CHANGES_REQUESTED** [review]
@jhuttana Please move the config of filenames to the top of the script so we have them in one place when we need to change them.

**Review by jerboaa — APPROVED** [review]
LGTM. We'll merge this once the 8u262 EA builds cycle starts.

**Comment by jerboaa** [comment]
jdk8u262-b01 got tagged. Merging this now.

**Pull request merged**